packages/base-driver/lib/basedriver/commands/execute.js

```diff
@@ -0,0 +1,58 @@
+// @ts-check
+import _ from 'lodash';
+import {makeArgs, checkParams} from '../../protocol';
+
+/**
+ * @param {SessionBase} Base
+ * @returns {ExecuteBase}
+ */
+export function ExecuteMixin(Base) {
+  /**
+   * @implements {IExecuteCommands}
+   */
+  class ExecuteCommands extends Base {
+    /**
+     * @this {Driver}
+     * @param {string} script
+     * @param {Array<Record<string, any>>} protoArgs
```

Comment:
Given the comment and code below,

```suggestion
     * @param {[Record<string, any>]|[]} protoArgs
```

...is more appropriate.  This is a "tuple" (a JS array having a specific number and order of items) with one member of type `Record<string,any>` _or_ an empty array.  It cannot be `[Record<string,any>|undefined]` because that would be an array having a single member `undefined`. It cannot be `[Record<string,any>?]` because that's another way of writing `[Record<string,any>|null]`, then the previous sentence applies.

Reply:
ah, good thinking!